### ISSUE TITLE
Regression Tests

### ISSUE THREAD

**Issue by peisenha** [grmToolbox/grmpy#11]
We need to integrate a regression test battery in our workflow. See my first draft at https://github.com/grmToolbox/grmpy/blob/master/development/tests/regression/draft.py
We will discuss the ideas behind it and the next steps during our call today.

**Comment by peisenha** [comment]
I just had a first look at your current implementation. Please incorporate the following changes, if they remain unclear please just let me know.

-  We want to have the regression test creation outside the package test directory. Please move that part fully to development/tests/regeression. The idea is then to create the json file there and simply copy it over to a to be created grmpy/test/resources directory.
- Just select a random subset of 5 tests to run inside the test_integreation/test_2, which reads the vault from the resources subdirectory

So, in a nutshell, keep the creation and running of regression tests more separated.

Let me know once you had another stab at it, have a good weekend!

**Comment by SeBecker** [comment]
I uploaded a new version to the repo. I hope that the files are more in accordance with your expectations as before. I was not sure if i was also supposed to integrate a loop in /test_integration/test_2. Just let me know if this should be the case.

**Comment by peisenha** [comment]
Thank you, the basic ideas are all there now. Just some small edits and extensions:

`development/tests/regression`

- When new tests are created, please do not directly copy them into the package's test folder. Instead keep them in the current directory. I want these to be manually copied only. Otherwise they might be updated by accident.

- No automatic testdir creation, that is part of the package from now on.

`grmpy/test/test_integration`

- Please practice uploading the package to pypi, see slack in grmpy channel. Now that we have non-code files (the regression vault), please amend the packaging files accordingly, see [here](http://python-packaging.readthedocs.io/en/latest/non-code-files.html). Upload the package, install it in a fresh virtual environment, and try to run `python -c "import grmpy; grmpy.test()"` our test suite. I am not sure that the way you hard-coded the directories will work.... See the respy package how I solved this if you do not find a better solution.

**Comment by peisenha** [comment]
Please add a brief explanation about regression tests to http://grmpy.readthedocs.io/en/latest/software_engineering.html along the lines from the respy package.

**Comment by SeBecker** [comment]
I changed the paths in the test_integration file for test2 and adjusted the regression battery. 
```python -c "import grmpy;grmpy.test()"``` works now if you download the package via pip in a fresh enviroment.

**Comment by peisenha** [comment]
Great, works like a charm. Please update the documentation and then assign the issue to me.

**Comment by peisenha** [comment]
Thanks for the update to the doc, I will work on it a bit more during the Hackathon on Thursday ...
